src/js/interpreter_cache.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "./interpreter_cache_interface.h"
+#include "ccf/pal/locking.h"
+#include "ds/lru.h"
+
+namespace ccf::js
+{
+  class InterpreterCache : public AbstractInterpreterCache
+  {
+  protected:
+    // Locks access to all internal fields
+    ccf::pal::Mutex lock;
+    LRU<std::string, std::shared_ptr<js::Context>> lru;
+    size_t cache_build_marker;
+
+  public:
+    InterpreterCache() : lru(10) {}
+
+    std::shared_ptr<js::Context> get_interpreter(
+      js::TxAccess access,
+      const JSDynamicEndpoint& endpoint,
+      size_t freshness_marker) override
+    {
+      if (access != js::TxAccess::APP)
+      {
+        throw std::logic_error(
+          "JS interpreter reuse lru is currently only supported for APP "
```

Comment:
```suggestion
          "JS interpreter reuse lru is only supported for APP "
```

I don't believe this would be a good tradeoff for governance, let's not imply we intend to make it available there until we have decided so.